libs/nextjs/src/helpers.ts

```diff
@@ -139,13 +123,82 @@ export async function refreshToken(
   }
 }
 
-export function addToCookies(cookieValue: string, res: ServerResponse) {
+type CreateCookieArguments = {
+  cookieName?: string,
+  session: string,
+  expires: CookieSerializeOptions['expires'],
+  isSecured: CookieSerializeOptions['secure'],
+  cookieDomain?: CookieSerializeOptions['domain'],
+  httpOnly?: CookieSerializeOptions['httpOnly'],
+  path?: CookieSerializeOptions['path']
+}
+const COOKIE_MAX_LENGTH = 4096
+export function createCookie(
+  {
+    cookieName = fronteggConfig.cookieName,
+    session,
+    expires,
+    isSecured,
+    cookieDomain = fronteggConfig.cookieDomain,
+    httpOnly = true,
+    path = '/'
+  }: CreateCookieArguments) {
+  const options = {
+    expires,
+    httpOnly,
+    domain: cookieDomain,
+    path,
+    sameSite: isSecured ? 'none' as const : undefined,
+    secure: isSecured,
+  }
+  const cookieValue = cookie.serialize(cookieName, session, options);
+  if (cookieValue.length < COOKIE_MAX_LENGTH) {
+    return [cookieValue]
+  }
+  return createSplitCookie(cookieName, session, options, cookieValue.length)
+}
+
+function createSplitCookie(cookieName: string, session: string, options: CookieSerializeOptions, cookieLength: number) {
+  const numberOfCookies = Math.ceil(cookieLength / COOKIE_MAX_LENGTH)
+  const splitSession = chunkString(session, numberOfCookies)
+  const allCookies = []
+  for (let i = 1; i <= numberOfCookies; i++) {
+    allCookies.push(cookie.serialize(`${cookieName}-${i}`, splitSession[i - 1], options))
+  }
+  return allCookies
+}
+
+function chunkString(str: string, numChunks: number) {
+  const chunkSize = Math.ceil(str.length / numChunks)
+  const chunks = []
+  for (let i = 0; i < numChunks; i + chunkSize) {
+    const limit = i + chunkSize
+    chunks.push(str.substring(i, limit < str.length ? limit : str.length))
+  }
+  return chunks
+}
+
+export function parseCookie(cookieStr: string) {
+  let sealFromCookies = '';
+  if (cookie.parse(cookieStr)[fronteggConfig.cookieName]) {
+    sealFromCookies = cookie.parse(cookieStr)[fronteggConfig.cookieName]
+  } else {
+    let i = 1;
+    while (cookie.parse(cookieStr)[`${fronteggConfig.cookieName}-${i}`]) {
+      sealFromCookies += cookie.parse(cookieStr)[`${fronteggConfig.cookieName}-${i}`]
+      i++;
+    }
+  }
+  return sealFromCookies !== '' ? sealFromCookies : undefined
```

Comment:
`return sealFromCookies || undefined` might be better? Also I see that returning just `return sealFromCookies` might also work.

Reply:
I will fix it the next pr